### ISSUE TITLE
Where is the rest of the material struct bound?

### ISSUE THREAD

**Issue by xSNOWM4Nx** [Microsoft/DirectX-Graphics-Samples#101]
I looked at the MiniEngine app. It provides a good introduction to DX12. Very cool ;-)
But it is not entirely clear to me where the rest of the material struct (vectors of ambient and emissive colors, floats like opacity, shininess etc.) is bound to the shaders. I can only see the textures as SRV's in the Model.h/ModelH3D.cpp.

**Comment by stanard** [comment]
Hi, I'm glad you like MiniEngine!  I apologize if ModelViewer's name is a bit misleading.  It's not a very general purpose model viewer.  It might be better called the SponzaViewer, because it's tailored to that specific scene.  The shaders we render with are designed around the data we found in the Sponza OBJ file.  We don't have a more generalized scene representation or a variety of materials to go with it.  But that shouldn't stop you from being able to write your own shaders and constant buffers to define your own materials with variable parameters.  So far, we've spent most of our time creating widely applicable and generally reusable code.  When it comes to defining your particular rendering style and materials, that's where people often go with something custom.  Cheers!
